app/controllers/collection_numbers_controller.rb

```diff
@@ -258,11 +267,12 @@ def figure_out_where_to_go_back_to
   end
 
   def redirect_to_observation_or_collection_number
-    if @back_object
-      redirect_with_query(@back_object.show_link_args)
+    if @back_object.is_a?(CollectionNumber)
```

Comment:
For some reason `show_link_args` is returning very confused args here, controller/action combos that don't make sense. I've encountered this problem also on other controller refactors and after trying to debug it for hours and hours, I'm more inclined to abandon it. I'm converting `show_link_args` to the more legible path helpers wherever I can.

If the controllers all had resource routes it could be simple to have `show_link_args` just send to the appropriate path helper method using string interpolation... and a `.controller_normalized?` check. However, it seems more helpful here to be reminded that there are only two kinds of `@back_object` that a collection number would send you to. It took me a few days to notice that myself.